### DISCUSSION TITLE
tsconfig.json path/filename

### DISCUSSION THREAD

**Discussion by dkonsoftware** [adonisjs/core#1579] [Help] [answered]
Hi is it possible to override tsconfig.json path or file name?

**Comment by thetutlage** [comment]
What's the use case?

**Reply by dkonsoftware** [reply]
- I want to integrate adonis with next.js (typescript) but as I know next doesn't support custom tsconfig path.
- I want to avoid managing workspaces/monorepo.
- I want to share DTOs between backend and frontend app.
- I want to avoid using "custom tsconfig path" plugins or special configuration in for eg. Inertia.js(typescript) setup.

**Reply by thetutlage** [reply]
I am still unsure how all this becomes possible by customizing the tsconfig file name. Especially these two

> I want to share DTOs between backend and frontend app.
> I want to avoid using "custom tsconfig path" plugins or special configuration in for eg. Inertia.js(typescript) setup.

Also, please do not assume that I am aware of how Next or Monorepos work. So you have to be explicit with what exactly you are trying to achieve and how the fixed path or name of the `tsconfig` file is becoming an issue.  Also, how you are planning to tell your code editor to use a different filename?

**Reply by dkonsoftware** [reply]
I want to simplify somehow frontend (typescript) app setup in adonis.
Since adonis 5 is typescript app we need to manage two different tsconfig.json files one for backend one for frontend.
When I use standard nextjs setup ("next" command) I have to use tsconfig.json file in root, for Inertia (typescript) setup I have to use special plugin for custom tsconfig path.

> Also, how you are planning to tell your code editor to use a different filename? 

Im thinking about setup tsconfig.json for frontend app which extends backend for eg. tsconfig.server.json file.
I'm also not sure if above will work since I don't have much experience with Typescript but definitely want to use it for full stack. Maybe you have better solution :) Thanks.

**Comment by dkonsoftware** [comment, marked as answer]
Hi if its helpful for someone and want to use TS on frontend as well as backend you can use [adonis-mix-asset](https://github.com/wahyubucil/adonis-mix-asset) package and solution for TypeScript. It works fine for me.
